docs/user/custom-domains.rst

```diff
@@ -2,125 +2,41 @@ Custom Domains
 ==============
```

Comment:
Overall this page is a good upgrade, but I do feel like we're talking a lot about canonical domains, but not in a super clear way. We should probably have something like a Canonical domain section, and a bullet list of what it enables, which I think would be clearer?

Reply:
We also have this open - do we need to solve it in order to reference it?

![image](https://user-images.githubusercontent.com/374612/204396976-18f7e9cd-7772-4889-aae3-d7f87a25b8cc.png)

https://github.com/readthedocs/readthedocs.org/issues/9746

Reply:
I don't quite follow? If our goal is to talk about canonical URLs here so much, then it does seem to have some overlap with those pages, so refactoring them in the same PR would make sense.